lib/edge_impulse/CMakeLists.txt

```diff
@@ -0,0 +1,138 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+
+if(CONFIG_EDGE_IMPULSE_ZIP_PATH STREQUAL "")
+  message(FATAL_ERROR "Edge Impulse archive path must be specified.")
+endif()
+
+# Set full path for Edge Impulse archive and base directory
+set(EI_ZIP_PATH ${APPLICATION_SOURCE_DIR}/${CONFIG_EDGE_IMPULSE_ZIP_PATH})
+set(EI_BASE_DIR ${CMAKE_BINARY_DIR}/ei_src_unpacked)
+
+# Download Edge Impulse library (if needed)
+if(NOT CONFIG_EDGE_IMPULSE_DOWNLOAD_URI STREQUAL "")
+  if(NOT EXISTS ${EI_ZIP_PATH})
+    # Start download, use API key that was provided.
+    message(STATUS "Downloading Edge Impulse files, please wait...")
+
+    if(DEFINED EI_API_KEY_HEADER)
+      file(DOWNLOAD ${CONFIG_EDGE_IMPULSE_DOWNLOAD_URI} ${EI_ZIP_PATH}
```

Comment:
Then the *.zip file will be re-downloaded every time when we clean the build directory. The current approach lets user to decide if the *.zip file should be updated.
That gives user more control on the zip content (the zip file located under the URI may change) and reduces number of requests sent to external HTTP server.

Reply:
But it pollutes the repo by adding untracked files. Judging from my experience the question is not if but when this thing will be committed ;)
User can always copy the zip and update the path if this is what is desired.
BTW - how big is the package?

Reply:
Package is around 4-5 MB. User could simply add .gitignore file in the application directory to prevent commiting the *.zip file (I did it for the sample locally).

Reply:
> User could simply add .gitignore file

Or accidently commit the package. The dependency on what user will do it the problem here.
I don't think a few MB package is a problem to download when build directory is cleaned.

Reply:
Changed. I also updated Kconfig options.